src/query/settings/src/lib.rs

```diff
@@ -105,13 +108,26 @@ impl Settings {
         // Overwrite settings from conf or global set.
         {
             // Set max threads.
-            if ret.get_max_threads()? == 0 {
-                let cpus = if conf.query.num_cpus == 0 {
-                    num_cpus::get() as u64
-                } else {
-                    conf.query.num_cpus
-                };
-                ret.set_max_threads(cpus)?;
+            {
+                if ret.get_max_threads()? == 0 {
+                    let cpus = if conf.query.num_cpus == 0 {
+                        sysinfo::System::new_with_specifics(
+                            RefreshKind::new().with_cpu(CpuRefreshKind::everything()),
```

Comment:
num_cpus has a physical cpu num method.

Reply:
I am not sure the physical CPU is right for some containers.

Reply:
We used `num_cpus::get()` as available CPUs, not physical cores.